File: epi_judge_python/look_and_say.py
from test_framework import generic_test
from itertools import groupby


def look_and_say(n: int) -> str:
    # res = "1"
    # for _ in range(n - 1):
    #     temp = ""
    #     digit = res[0]
    #     count = 0
    #     for c in res:
    #         if c == digit:
    #             count += 1
    #         else:
    #             temp += str(count) + digit
    #             count = 1
    #             digit = c
    #     temp += str(count) + digit
    #     res = temp
    # return res
    res = "1"
    for _ in range(n - 1):
        temp = ""
        for k, g in groupby(res):
            temp += str(len(list(g))) + k
        res = temp
    return res


if __name__ == "__main__":
    exit(
        generic_test.generic_test_main(
            "look_and_say.py", "look_and_say.tsv", look_and_say
        )
    )

File: epi_judge_cpp/look_and_say.cc
#include <string>

#include "test_framework/generic_test.h"
using std::string;

string LookAndSay(int n) {
  string current{"1"};
  for (auto i = 1; i < n; i++) {
    string prev(current);
    current.clear();
    char seen = prev[0];
    int count = 0;
    for (auto &ch : prev) {
      if (ch == seen) {
        count++;
      } else {
        current += std::to_string(count);
        current.push_back(seen);
        seen = ch;
        count = 1;
      }
    }
    if (count) {
      current += std::to_string(count);
      current.push_back(seen);
    }
  }
  return current;
}

int main(int argc, char *argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"n"};
  return GenericTestMain(args, "look_and_say.cc", "look_and_say.tsv",
                         &LookAndSay, DefaultComparator{}, param_names);
}
